src/Functions/date_trunc.cpp

```diff
@@ -187,7 +187,75 @@ class FunctionDateTrunc : public IFunction
 
 REGISTER_FUNCTION(DateTrunc)
 {
-    factory.registerFunction<FunctionDateTrunc>();
+    FunctionDocumentation::Description description = R"(
+Truncates a date and time value to the specified part of the date.
+    )";
+    FunctionDocumentation::Syntax syntax = R"(
+dateTrunc(unit, datetime[, timezone])
+    )";
+    FunctionDocumentation::Arguments arguments = {
+        {"unit", R"(The type of interval to truncate the result. `unit` argument is case-insensitive. [`String`](/sql-reference/syntax#string).
+
+| Unit         | Compatibility                   |
+|--------------|---------------------------------|
+| `nanosecond` | Compatible only with DateTime64 |
+| `microsecond`| Compatible only with DateTime64 |
+| `millisecond`| Compatible only with DateTime64 |
+| `second`     |                                 |
+| `minute`     |                                 |
+| `hour`       |                                 |
+| `day`        |                                 |
+| `week`       |                                 |
+| `month`      |                                 |
+| `quarter`    |                                 |
+| `year`       |                                 |
+)"},
+        {"datetime", "Date and time. [`Date`](../data-types/date.md) or [`Date32`](../data-types/date32.md) or [`DateTime`](../data-types/datetime.md) or [`DateTime64`](../data-types/datetime64.md)."},
+        {"timezone", "Optional. Timezone name for the returned datetime. If not specified, the function uses the timezone of the `datetime` parameter. [`String`](../data-types/string.md)."}
+    };
+    FunctionDocumentation::ReturnedValue returned_value = R"(
+Returns the truncated date and time value.
+
+| Unit Argument               | `datetime` Argument     | Return Type                                      |
+|-----------------------------|-------------------------|--------------------------------------------------|
+| Year, Quarter, Month, Week  | Date32 or DateTime64    | [`Date32`](../data-types/date32.md)              |
+| Year, Quarter, Month, Week  | Date or DateTime        | [`Date`](../data-types/date.md)                  |
+| Day, Hour, Minute, Second   | Date32 or DateTime64    | [`DateTime64`](../data-types/datetime64.md)      |
+| Day, Hour, Minute, Second   | Date or DateTime        | [`DateTime`](../data-types/datetime.md)          |
+| Millisecond, Microsecond,   | Any                     | [`DateTime64`](../data-types/datetime64.md)      |
```

Comment:
It would be nice to make l. 225 and 226 a single row, otherwise one could think these are separate cases.